go/adbc/driver/internal/shared_utils.go

```diff
@@ -246,27 +292,63 @@ func (g *GetObjects) appendDbSchema(catalogName, dbSchemaName string) {
 	}
 	g.dbSchemaTablesBuilder.Append(true)
 
-	for _, tableInfo := range g.tableLookup[CatalogAndSchema{
-		Catalog: catalogName,
-		Schema:  dbSchemaName,
-	}] {
-		g.appendTableInfo(tableInfo)
+	catalogAndSchema := CatalogAndSchema{Catalog: catalogName, Schema: dbSchemaName}
+	for _, tableInfo := range g.tableLookup[catalogAndSchema] {
+		g.appendTableInfo(tableInfo, catalogAndSchema)
 	}
 }
 
-func (g *GetObjects) appendTableInfo(tableInfo TableInfo) {
+func (g *GetObjects) appendTableInfo(tableInfo TableInfo, catalogAndSchema CatalogAndSchema) {
 	g.tableNameBuilder.Append(tableInfo.Name)
 	g.tableTypeBuilder.Append(tableInfo.TableType)
 	g.dbSchemaTablesItems.Append(true)
 
+	if len(g.ConstraintLookup) != 0 {
+		g.tableConstraintsBuilder.Append(true)
+
+		catalogSchemaTable := CatalogSchemaTable{Catalog: catalogAndSchema.Catalog, Schema: catalogAndSchema.Schema, Table: tableInfo.Name}
+		constraintSchemaData, exists := g.ConstraintLookup[catalogSchemaTable]
+
+		if exists {
+			for _, data := range constraintSchemaData {
+				g.constraintNameBuilder.Append(data.ConstraintName)
+				g.constraintTypeBuilder.Append(data.ConstraintType)
+
+				if len(data.ConstraintColumnNames) == 0 {
```

Comment:
Any chance we can handle these more cleanly? Like adding a function to handle lists or append null if empty that takes in a lambda? (Not a Go expert).

If this were C# you could do an extension method on the builder but not sure if the concept exists in Go (or perhaps just a new instance method on the builder...).

Reply:
modularized the code, by refactoring the logic of table constraints into methods.